_studio/mfx_lib/shared/src/mfx_mpeg2_enc_common_hw.cpp

```diff
@@ -62,7 +62,14 @@ mfxStatus MfxHwMpeg2Encode::QueryHwCaps(VideoCORE* pCore, ENCODE_CAPS & hwCaps,
     ddi.reset( CreatePlatformMpeg2Encoder(pCore) );
     if(ddi.get() == NULL)
         return MFX_ERR_NULL_PTR;
-    mfxStatus sts = ddi->QueryEncodeCaps(hwCaps, codecProfileType);
+
+    mfxU16 codecProfile = par->mfx.CodecProfile;
```

Comment:
Please either pass "par" by const& or check pointer for 0 before dereference. Better to pass using const& because it's not expected that "par" can be zero.

Reply:
Not need if we define profile at higher levels of the stack